src/source/file_system.rs

```diff
@@ -11,7 +11,17 @@ pub enum DirEntryKind {
   File,
   Dir,
   Symlink,
-  Error(anyhow::Error),
+  Error(DirEntryError),
+}
+
+#[derive(Debug, thiserror::Error)]
+pub enum DirEntryError {
+  #[error("Failed to read directory.")]
+  Dir(std::io::Error),
+  #[error("Failed to read file type.")]
+  FileType(std::io::Error),
+  #[error("Failed to read entry in directory.")]
+  DirEntry(std::io::Error),
```

Comment:
Do these need to have `#[source]` to say what the error is? Or will thiserror do that automatically?

Reply:
ah good point, will add that